readthedocs/restapi/views/model_views.py

```diff
@@ -215,6 +215,17 @@ def sync_versions(self, request, **kwargs):  # noqa: D205
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
+        try:
+            api_utils.run_automation_rules(project, added_versions)
```

Comment:
The order of the versions inside `added_versions` will have an impact in which version will be the `default_version` in the end if there are multiple matches.

I'm not sure what's best here, maybe we want to sort the versions applying some rule?

Reply:
I was thinking about this, and there are some cases to consider. But I think (and hope) those are not going to be very common.

I think it's not very common to receive webhook with more than one branch/tag (or both). At least that someone is pushing from a local repo at once a lot of tags and branches.

Going back the order, we lose the order in the very beginning, when we separate branches from tags, and also, we don't keep track of the commit time.

I think we could keep track of the commit time and improve this, and maybe it will be useful for another stuff (like sorting versions).

I'm going to create an issue.

Reply:
I think it's a good idea to write a comment in the code explaining this possible situation so it's not a surprise when debugging this strange case.